ext/pdo_sqlite/sqlite_statement.c

```diff
@@ -387,21 +392,78 @@ static int pdo_sqlite_stmt_get_attribute(pdo_stmt_t *stmt, zend_long attr, zval
 				ZVAL_TRUE(val);
 			}
 			break;
+        case PDO_SQLITE_ATTR_EXPLAIN_STATEMENT:
+#if SQLITE_VERSION_NUMBER >= 3041000
+#if defined(__APPLE__)
+            if (__builtin_available(macOS 14.2, *)) {
+#endif
+                ZVAL_LONG(val, (zend_long)sqlite3_stmt_isexplain(S->stmt));
+                return 1;
+#if defined(__APPLE__)
+            } else {
+                zend_value_error("explain statement unsupported");
+                return 0;
+            }
+#endif
+#else
+            zend_value_error("explain statement unsupported");
+            return 0;
+#endif
 		default:
 			return 0;
 	}
 
 	return 1;
 }
 
+static int pdo_sqlite_stmt_set_attribute(pdo_stmt_t *stmt, zend_long attr, zval *zval)
+{
+	pdo_sqlite_stmt *S = (pdo_sqlite_stmt*)stmt->driver_data;
+
+	switch (attr) {
+        case PDO_SQLITE_ATTR_EXPLAIN_STATEMENT:
+#if SQLITE_VERSION_NUMBER >= 3041000
+#if defined(__APPLE__)
+            if (__builtin_available(macOS 14.2, *)) {
+#endif
+                if (Z_TYPE_P(zval) != IS_LONG) {
+                    zend_type_error("explain mode must be of type int, %s given", zend_zval_value_name(zval));
+                    return 0;
+                }
+                if (Z_LVAL_P(zval) < 0 || Z_LVAL_P(zval) > 2) {
+                    zend_value_error("explain mode must be one of the EXPLAIN_MODE_* constants");
```

Comment:
For class constants we normally prefix the class name, i.e. `Pdo\Sqlite::EXPLAIN_MODE_* constants`